src/OrchardCore.Modules/OrchardCore.UrlRewriting/ViewModels/RewriteRuleViewModel.cs

```diff
@@ -0,0 +1,63 @@
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace OrchardCore.UrlRewriting.ViewModels;
+
+public class RewriteRuleViewModel
+{
+    public string Name { get; set; }
+
+    public string Pattern { get; set; }
+
+    public bool IgnoreCase { get; set; }
+
+    public RuleAction RuleAction { get; set; }
+
+    public RewriteActionViewModel RewriteAction { get; set; } = new RewriteActionViewModel();
+
+    public RedirectActionViewModel RedirectAction { get; set; } = new RedirectActionViewModel();
```

Comment:
having a vm as a property of another vm seems wired to me. we'll see below with the rest of the code